gh-pages-stub/public/js/snippets.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+var snipArr, arrl, visibleArr, wholeArr;
+snipArr = document.getElementsByTagName("CODE");
+arrl = snipArr.length;
+visibleArr = [];
+wholeArr = [];
+cutAll();
+
+function cutAll() {
+  var i, delim;
+  for (i=0; i<arrl; i++) {
+    delim = snipArr[i].innerHTML.indexOf("&gt;&gt;&gt;showmore");
+    wholeArr[i] = snipArr[i].innerHTML;
+    if (delim === -1) {continue;}
+    visibleArr[i] = snipArr[i].innerHTML.slice(0,delim);
+    snipArr[i].parentElement.innerHTML += "<button class='collapse_control more collapsed' id='showmoreId" + i + "' onclick='showMore()'>Show more</button>";
+    snipArr[i].innerHTML = visibleArr[i];
+  }
+}
+
+function showMore() {
+  var x, y, z, start, end, whole, parentPre;
+  parentPre = event.target.parentElement;
+  x = event.target.id.slice(10);
+  y = wholeArr[x].indexOf("&gt;&gt;&gt;showmore");
+  start = wholeArr[x].slice(0,y);
+  z = y + 21;
```

Comment:
Some black magic is happening here.

Reply:
Lolz, 21 is the delimiter length. Fixed.